erigon-lib/rlp/encode.go

```diff
@@ -725,3 +725,66 @@ func EncodeStructSizePrefix(size int, w io.Writer, buffer []byte) error {
 	}
 	return nil
 }
+
+func ByteSliceSliceSize(bb [][]byte) int {
+	size := 0
+	for i := 0; i < len(bb); i++ {
+		size += StringLen(bb[i])
+	}
+	return size + ListPrefixLen(size)
+}
+
+func EncodeByteSliceSlice(bb [][]byte, w io.Writer, b []byte) error {
+	totalSize := 0
+	for i := 0; i < len(bb); i++ {
+		totalSize += StringLen(bb[i])
+	}
+
+	if err := EncodeStructSizePrefix(totalSize, w, b); err != nil {
+		return err
+	}
+
+	for i := 0; i < len(bb); i++ {
+		if err := EncodeString(bb[i], w, b); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func ByteSliceSlicePtrSize(bb []*[]byte) int {
```

Comment:
Why do we need this? To my mind it doesn't make sense to have pointer to a slice since slices [have reference semantics](https://stackoverflow.com/questions/72978660/is-it-correct-to-use-slice-as-item-because-slice-is-by-default-pointer).